lib/rspec/rails/matchers/base_matcher.rb

```diff
@@ -0,0 +1,183 @@
+module RSpec
+  module Rails
+    module Matchers
+      class BaseMatcher
```

Comment:
You should tag this as `@api private` so it's clear this we're not considering this part of our public API subject to SemVer.